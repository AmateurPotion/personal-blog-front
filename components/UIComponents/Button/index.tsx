import * as React from "react";
import styled from 'styled-components';

interface Props{
    children?: string;
    onClick?(): void;
};

const Button : React.SFC<Props> = (props) => {
    const {children, onClick} = props;
    return (
    <>
        <PlusButton onClick={onClick}>{children}</PlusButton>
    </>
    );
};

Button.defaultProps = {
    children: "DEFAULT",
    onClick:() => {}
}

const PlusButton = styled.div`
    display: inline-block;
    text-align: center;
    margin:2rem auto;
    padding: 1rem;
    background: linear-gradient(45deg, #e45d4c,#eead9e);
    border-radius: 3rem;
    font-weight: 700;
    color: white;
    cursor:pointer;
`;

export default Button;