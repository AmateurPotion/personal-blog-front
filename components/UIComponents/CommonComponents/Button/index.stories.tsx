/**
 * After storybook v5.3, They recommend use CSF(Component Story Format)
 * instead of using storiesof API
 */
import React from "react";
import Component from "./index";
import { withKnobs, text } from "@storybook/addon-knobs";
import centered from "@storybook/addon-centered";

export default {
    title: "COMPONENTS|Common/Button",
    component: Component,
    decorators: [withKnobs, centered],
    parameter: {
        componentSubtitle: "Global common botton"
    }
};

export const standard = () => {
    const title = text("buttonText", "DEFAULT");
    return <Component>{title}</Component>;
};
export const disabled = () => {
    const title = text("buttonText", "DEFAULT");
    return <Component disabled={true}>{title}</Component>;
};
