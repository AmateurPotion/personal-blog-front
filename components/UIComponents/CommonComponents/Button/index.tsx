import React from "react";
import styled from "styled-components";

interface IProps {
    children?: string;
    onClick?(): void;
    disabled?: boolean;
}

/**
 * 안녕하세요 라고 보여주고 싶을 땐 `Hello` 컴포넌트를 사용하세요.
 *
 * - `big` 값을 `true`로 설정하면 **크게** 나타납니다.
 * - `onHello` 와 `onBye` props로 설정하여 버튼이 클릭했을 때 호출 할 함수를 지정 할 수 있습니다.
 */
const Button: React.FC<IProps> = props => {
    const { children, onClick, disabled } = props;
    return (
        <>
            <PlusButton onClick={onClick} disabled={disabled}>
                {children}
            </PlusButton>
        </>
    );
};

Button.defaultProps = {
    children: "DEFAULT",
    onClick: () => {},
    disabled: false
};

const PlusButton = styled.div<IProps>`
    display: inline-block;
    text-align: center;
    margin: 2rem auto;
    padding: 0.8rem 1rem;
    background: ${props => (props.disabled ? props.theme.achromaticColor : `linear-gradient(45deg, #e45d4c,#eead9e)`)};
    border-radius: 1.5rem;
    font-weight: 700;
    color: white;
    cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
`;

export default Button;
