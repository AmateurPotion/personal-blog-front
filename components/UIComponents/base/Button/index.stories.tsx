/**
 * After storybook v5.3, They recommend use CSF(Component Story Format)
 * instead of using storiesof API
 */
import React from "react";
import Component from "./index";
import { withKnobs, text } from "@storybook/addon-knobs";

export default {
    title: "COMPONENTS|Base/Button",
    component: Component,
    decorators: [withKnobs],
    parameter: {
        componentSubtitle: "Global common botton"
    }
};

export const standard = () => {
    const title = text("buttonText", "DEFAULT");
    return <Component>{title}</Component>;
};

export const disabled = () => {
    const title = text("buttonText", "DEFAULT");
    return <Component disabled={true}>{title}</Component>;
};
