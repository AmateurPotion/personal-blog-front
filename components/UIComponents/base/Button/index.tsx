import React from "react";
import styled from "styled-components";

interface IProps {
    children?: string;
    onClick?: (event) => void;
    disabled?: boolean;
}

const Button: React.FC<IProps> = props => {
    const { children, onClick, disabled } = props;

    return (
        <Container onClick={onClick} disabled={disabled}>
            {children}
        </Container>
    );
};

Button.defaultProps = {
    children: "DEFAULT",
    onClick: () => {},
    disabled: false
};

const Container = styled.div<IProps>`
    display: inline-block;
    color: white;
    font-weight: 700;
    text-align: center;
    margin: 2rem auto;
    padding: 0.8rem 1rem;
    background: ${props => (props.disabled ? props.theme.color.achromatic : props.theme.color.gradient)};
    border-radius: ${props => props.theme.BORDER_RADIUS};
    cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
`;

export default Button;
