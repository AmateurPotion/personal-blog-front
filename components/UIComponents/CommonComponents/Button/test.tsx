import * as React from "react";
import Button from "./index";
import renderer from "react-test-renderer";

test("Button Component Rendering", () => {
    const component = renderer.create(<Button>Default</Button>);
    const domTree = component.toJSON();
    expect(domTree).toMatchSnapshot();
});
