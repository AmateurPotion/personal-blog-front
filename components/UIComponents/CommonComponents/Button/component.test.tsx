import React from "react";
import Button from "./index";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("[Button] Component Rendering", () => {
    it("should render without crashing", () => {
        const component = renderer.create(<Button>Default</Button>);
        const domTree = component.toJSON();
        expect(domTree).toMatchSnapshot();
    });
    
    it("should render disabled without crashing", () => {
        const component = renderer.create(<Button disabled={true}>Default</Button>);
        const domTree = component.toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('defualt value is "DEFAULT"', () => {
        const component = shallow(<Button />);
        expect(component.text()).toEqual("DEFAULT");
    });

    it('rendering Text well with "로그인 버튼"', () => {
        const component = shallow(<Button>로그인 버튼</Button>);
        expect(component.text()).toEqual("로그인 버튼");
    });
});
