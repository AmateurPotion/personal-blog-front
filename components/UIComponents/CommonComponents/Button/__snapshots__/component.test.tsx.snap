// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] Component Rendering should render disabled without crashing 1`] = `
<div
  className="sc-bdVaJa YfLeS"
  disabled={true}
  onClick={[Function]}
>
  Default
</div>
`;

exports[`[Button] Component Rendering should render without crashing 1`] = `
<div
  className="sc-bdVaJa bRgoTr"
  disabled={false}
  onClick={[Function]}
>
  Default
</div>
`;
